### PR TITLE
Add ProcessFinder

### PR DESCRIPTION
**Pull request by niusmallnan**
Then longhorn-manager and longhorn-engine will search `dockerd` by `ProcessFinder`.

The rough code will be:
```
procPath := getProcPath()
pf := new ProcessFinder(procPath)
ps, _  := pf.FindTarget("dockerd")
if ps == nil {
      logrus.Infof("Failed to search dockerd, pid1 will be used")
      return 1
}
return ps.Pid
```

### CODE REVIEW

**Comment by yasker** [comment]
```
procPath := getProcPath()
pf := new ProcessFinder(procPath)
ps, err := pf.FindTarget("dockerd")
if err != nil {
      logrus.Warnf("Failed to search dockerd, pid 1 will be used")
      return 1
}
return ps.Pid
```

File: util/util_test.go [hunk 1]
```diff
@@ -36,3 +36,15 @@ func (s *TestSuite) TestNamespaceExecutor(c *C) {
 	_, err = ne.Execute("mount", []string{})
 	c.Assert(err, IsNil)
 }
+
+// TestFindDockerdProcess can pass in dapper
+// ENV DAPPER_RUN_ARGS --privileged -v /dev:/host/dev -v /proc:/host/proc
```
Comment (yasker) [inline util/util_test.go]: Don't need the comment. `go-iscsi-helper` way of building is by `make`, not `go build && go test`. As long as `make` works, it's fine.

File: util/process.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	path := fmt.Sprintf("%s/%d/status", p.procPath, pid)
+	return ReadProcessStatus(path)
+}
+
+func (p *ProcessFinder) FindSelf() (*ProcessStatus, error) {
+	path := fmt.Sprintf("%s/self/status", p.procPath)
+	return ReadProcessStatus(path)
+}
+
+func (p *ProcessFinder) FindTarget(targetProcess string) (*ProcessStatus, error) {
```
Comment (yasker) [inline util/process.go]: Use the name `FindAncestorByName`

File: scripts/validate [hunk 3]
```diff
@@ -11,7 +11,7 @@ echo Running: go vet
 go vet ${PACKAGES}
 echo Running: golint
 for i in ${PACKAGES}; do
-    if [ -n "$(golint $i | grep -v 'should have comment.*or be unexported' | tee /dev/stderr)" ]; then
+    if [ -n "$(golint $i | grep -v 'should have comment.*or be unexported' | grep -v 'process_status.go' | tee /dev/stderr)" ]; then
```
Comment (yasker) [inline scripts/validate]: Remove this after import the package `c9s/goprocinfo`.

File: util/util_test.go [hunk 4]
```diff
@@ -36,3 +36,15 @@ func (s *TestSuite) TestNamespaceExecutor(c *C) {
...
+func (s *TestSuite) TestFindDockerdProcess(c *C) {
+	procPath := "/host/proc"
+	dockerdProcess := "dockerd"
+
+	finder := NewProcessFinder(procPath)
+	ps, err := finder.FindTarget(dockerdProcess)
+	c.Assert(err, IsNil)
+	c.Assert(ps, NotNil)
```
Comment (yasker) [inline util/util_test.go]: Add a negative test case.

**Comment by yasker** [comment]
LGTM. Merged, thanks.

**Comment by niusmallnan** [comment]
https://github.com/rancher/longhorn/issues/10

**Pull request merged**